Doc/library/asyncio-stream.rst

```diff
@@ -367,13 +366,177 @@ Stream
 
    Represents a Stream object that provides APIs to read and write data
    to the IO stream . It includes the API provided by :class:`StreamReader`
-   and :class:`StreamWriter`.
+   and :class:`StreamWriter`. It can also be used as :term:`asynchronous iterator`
+   where :meth:`readline` is used. It raises :exc:`StopAsyncIteration` when
+   :meth:`readline` returns empty data.
 
    Do not instantiate *Stream* objects directly; use API like :func:`connect`
    and :class:`StreamServer` instead.
 
    .. versionadded:: 3.8
 
+   .. attribute:: mode
+
+      Returns the mode of the stream which is a :class:`StreamMode` value.
```

Comment:
I think we need to elaborate on this.  Add a sentence to list the enum values & explain what they mean.

Reply:
Agree, mentioning *read*, *write* and *read-write* modes make sense.
*Read* and *write* are used for pipes only, sockets are always read-write.

Reply:
Thanks, I added a bullet list of the values and a sentence about them.